bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/projectview/BazelProjectView.java

```diff
@@ -51,7 +51,8 @@ public record BazelProjectView(
         boolean discoverAllExternalAndWorkspaceJars,
         GlobSetMatcher externalJarsDiscoveryFilter,
         boolean shardSync,
-        int targetShardSize) {
+        int targetShardSize,
+        int importDepth) {
```

Comment:
This optimization applies to only two target provisioning strategies. Thus, I think it should not be a separate thing in the project view but part of `target_provisioning_settings` map.
https://github.com/salesforce/bazel-eclipse/blob/main/docs/common/projectviews.md#target_provisioning_settings

Can you revert this and just use it from there?

Also, doc update should be part of this PR. The `docs/common/projectviews.md `is the proper place. It should also briefly mention the negative impacts, i.e. possible compile errors, lost ability to run unit tests, missing sources during debugging.